src/sentry/api/bases/avatar.py

```diff
@@ -30,12 +31,36 @@ class SentryAppLogoSerializer(serializers.Serializer):
     avatar_type = serializers.ChoiceField(choices=(("default", "default"), ("upload", "upload")))
     color = serializers.BooleanField(required=True)
 
+    def is_black_and_white(self, data):
```

Comment:
I know a triple nested for loop is not great but it's limited - `AvatarField` already checked the file size (and that it's a square image).

Reply:
You should move this function to `/src/sentry/utils/avatar/`.